File: LeetCode/Easy/0001-two-sum/Python3/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_list = {}
        answer = []

        for i, n in enumerate(nums):
            find = target - n

            if find in num_list:
                answer.append(num_list[find])
                answer.append(i)
            
            else:
                num_list[n] = i
        
        return answer

File: LeetCode/Easy/0001-two-sum/Java/0001-two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> num_list = new HashMap<>();
        List<Integer> idxs = new ArrayList<Integer>();

        for(int i=0; i<nums.length; i++) {
            int find = target - nums[i];

            if(num_list.containsKey(find)) {
                idxs.add(num_list.get(find));
                idxs.add(i);
            }
            else {
                num_list.put(nums[i], i);
            }
        }

        int[] answer = new int[idxs.size()];
        for(int i=0; i<idxs.size(); i++) {
            answer[i] = idxs.get(i);
        }

        return answer;
    }
}